### PR TITLE
Fuzz fixes

### PR DESCRIPTION
**Pull request by bpfoley**
Fixes for a number of memory leaks, uninitialized value reads, and integer overflows found by
fuzzing.

### CODE REVIEW

File: src/sas/readstat_sas7bdat_read.c [hunk 1]
```diff
@@ -172,6 +172,7 @@ static readstat_error_t sas7bdat_realloc_col_info(sas7bdat_ctx_t *ctx, size_t co
         if (ctx->col_info == NULL) {
             return READSTAT_ERROR_MALLOC;
         }
+        memset(ctx->col_info, 0, ctx->col_info_count * sizeof(col_info_t));
```
[the remark this answers is not in the document]
Comment (bpfoley) [reply src/sas/readstat_sas7bdat_read.c]: Oops. Thanks. Will fix.

File: src/spss/readstat_sav_parse.rl [hunk 2]
```diff
@@ -222,8 +222,13 @@ readstat_error_t sav_parse_very_long_string_record(void *data, int count, sav_ct
...
         
         action incr_val {
-            if (fc != '\0') { 
-                temp_val = 10 * temp_val + (fc - '0'); 
+            if (fc != '\0') {
+                unsigned char digit = fc - '0';
+                if (temp_val <= (INT_MAX - digit) / 10) {
+                    temp_val = 10 * temp_val + digit;
+                } else {
+                    retval = READSTAT_ERROR_PARSE;
```
[the remark this answers is not in the document]
Comment (bpfoley) [reply src/spss/readstat_sav_parse.rl]: Right. I think the keyword in question is fbreak. Going by the generated C, it breaks out of the FSM processing loop, and lets the subsequent error handling code report the error.

**Comment by bpfoley** [comment]
Closing, now that this has been split into separate PRs.

**Comment by evanmiller** [comment]
Looking good, merged most of them. Thank you for your work on this.

**Pull request closed without merging**